wurst/lib/Lodash.wurst

```diff
@@ -305,6 +315,199 @@ public function each<T>(VoidFunction<T> func, LinkedList<T> list)
 public function LinkedList<T>.each<T>(VoidFunction<T> func)
     each(func, this)
 
+public function zip<A, B>(LinkedList<A> a, LinkedList<B> b) returns IterableMap<A, B>
+    let map = new IterableMap<A, B>()
+    let iterA = a.iterator()
+    let iterB = b.iterator()
+    while iterA.hasNext()
+        let bVal = iterB.hasNext()
+            ? iterB.next()
+            : null
+        map.put(iterA.next(), bVal)
+    iterA.close()
+    iterB.close()
+    a.maybeFree()
+    b.maybeFree()
+    return map
+
+public function LinkedList<A>.zip<A, B>(LinkedList<B> lst) returns IterableMap<A, B>
+    return zip(this, lst)
+
+public class Range
+    let incr = 1
+    let min = 0
+    let max = INT_MAX
+    var curr = min
+
+    construct()
+
+    construct(int min, int max, int incr)
+        this.min = min
+        this.max = max
+        this.incr = incr
+        curr = min
+
+    function hasNext() returns bool
+        return curr <= max
+
+    function next() returns int
+        let ret = curr
+        curr += incr
+        return ret
+
+    function close()
+        destroy this
+
+    function iterator() returns Range
+        return new Range(min, max, incr)
+
+    function toList() returns LinkedList<int>
```

Comment:
Is there not general functionality for converting an iterator to a list?

Reply:
@jonathanmclaus not that I'm aware of